ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/PackageCommitterSpec.scala

```diff
@@ -3,69 +3,464 @@
 
 package com.daml.ledger.participant.state.kvutils.committer
 
+import java.util.UUID
+
 import com.codahale.metrics.MetricRegistry
-import com.daml.ledger.participant.state.kvutils.DamlKvutils.DamlPackageUploadEntry
-import com.daml.ledger.participant.state.kvutils.TestHelpers._
+import com.daml.daml_lf_dev.DamlLf
 import com.daml.ledger.participant.state.kvutils.Conversions.buildTimestamp
-import com.daml.lf.engine.Engine
+import com.daml.ledger.participant.state.kvutils.DamlKvutils._
+import com.daml.ledger.participant.state.kvutils.TestHelpers._
+import com.daml.ledger.participant.state.kvutils.committer.PackageCommitter._
+import com.daml.lf.archive.Decode
+import com.daml.lf.archive.testing.Encode
+import com.daml.lf.data.Ref
+import com.daml.lf.engine.{Engine, EngineConfig}
+import com.daml.lf.language.Ast
 import com.daml.metrics.Metrics
-import org.scalatest.mockito.MockitoSugar
-import org.scalatest.{Matchers, WordSpec}
+import com.google.protobuf.ByteString
+import org.scalatest.{Matchers, ParallelTestExecution, WordSpec}
 
-class PackageCommitterSpec extends WordSpec with Matchers with MockitoSugar {
-  private val metrics = new Metrics(new MetricRegistry)
-  private val anEmptyResult = DamlPackageUploadEntry.newBuilder
-    .setSubmissionId("an ID")
-    .setParticipantId("a participant")
+import scala.collection.JavaConverters._
 
-  "buildLogEntry" should {
-    "set record time in log entry if record time is available" in {
-      val instance = new PackageCommitter(mock[Engine], metrics)
-      val context = new FakeCommitContext(recordTime = Some(theRecordTime))
+class PackageCommitterSpec extends WordSpec with Matchers with ParallelTestExecution {
 
-      val actual = instance.buildLogEntry(context, anEmptyResult)
+  import com.daml.lf.testing.parser.Implicits._
 
-      actual match {
-        case StepContinue(_) => fail
-        case StepStop(actualLogEntry) =>
-          actualLogEntry.hasRecordTime shouldBe true
-          actualLogEntry.getRecordTime shouldBe buildTimestamp(theRecordTime)
-      }
+  private[this] def encodePackage[P](pkg: Ast.Package) =
+    Encode.encodeArchive(
+      defaultParserParameters.defaultPackageId -> pkg,
+      defaultParserParameters.languageVersion,
+    )
+
+  private[this] val archive1 =
```

Comment:
Can you name these archives so that the name describes e.g. what test case they represent, please? I.e., it's hard to follow tests referring to numbered variables with a generic name. If these archives don't need to be distinct then let's generate them using a simple loop.

Reply:
done